extensions/SharkPool/BetterInput.js

```diff
@@ -0,0 +1,1305 @@
+          this.showButton4 = action === "Enabled";
+          break;
+      }
+    }
+
+    setInputType(args) {
+      this.isInputEnabled = args.ACTION;
+    }
+
+    setButtonText(args) {
+      const buttonMenu = args.BUTTON_MENU;
+      const text = args.TEXT;
+      switch (buttonMenu) {
+        case "Button 1":
+          this.submitButtonText = text;
+          break;
+        case "Button 2":
+          this.cancelButtonText = text;
+          break;
+        case "Button 3":
+          this.Button3Text = text;
+          break;
+        case "Button 4":
+          this.Button4Text = text;
+          break;
+        case "Dropdown":
+          this.DropdownText = text;
+          break;
+      }
+    }
+
+    askAndWaitForInput(args) {
+      if (this.askBoxCount < this.maxBoxCount) {
+        return new Promise((resolve) => {
+          this.askAndWait(args).then(() => {
+            resolve(this.getUserInput());
+          });
+        });
+      }
+    }
+
+    removeAskBoxes() {
+      const askBoxes = document.querySelectorAll(".ask-box");
+      askBoxes.forEach((box) => {
+        box.parentNode.removeChild(box);
+      });
+
+      if (this.askBoxPromise) {
+        this.askBoxPromise.resolve("removed");
+        this.askBoxPromise = null;
+      }
+      this.askBoxCount = 0;
+      this.userInput = "";
+    }
+
+    applyEnterEffect(overlay) {
+      overlay.style.opacity = "1";
+      overlay.style.transform = "scale(1)";
+
+      let currentOpacity = 0;
+      let currentScale = 0;
+
+      if (this.enterEffect !== "None") {
+        if (this.enterEffect === "fadeIn" || this.enterEffect === "Both") {
+          overlay.style.opacity = "0";
+        }
+
+        if (this.enterEffect === "growIn" || this.enterEffect === "Both") {
+          overlay.style.transform = "scale(0)";
+          overlay.style.transition = "opacity 0.3s";
+        }
+
+        const step = () => {
+          if (
+            (this.enterEffect === "fadeIn" || this.enterEffect === "Both") &&
+            currentOpacity < 100
+          ) {
+            currentOpacity += this.enterSpeed;
+            overlay.style.opacity = `${currentOpacity}%`;
+          }
+
+          if (
+            (this.enterEffect === "growIn" || this.enterEffect === "Both") &&
+            currentScale < 100
+          ) {
+            currentScale += this.enterSpeed;
+            overlay.style.transform = `scale(${currentScale / 100})`;
+          }
+
+          if (
+            (this.enterEffect === "Both" &&
+              (currentOpacity < 100 || currentScale < 100)) ||
+            (this.enterEffect === "growIn" && currentScale < 100) ||
+            (this.enterEffect !== "Both" && currentOpacity < 100)
+          ) {
+            requestAnimationFrame(step);
+          }
+        };
+
+        if (this.enterEffect === "growIn" || this.enterEffect === "Both") {
+          overlay.style.transition = "opacity 0.3s";
+        }
+
+        requestAnimationFrame(step);
```

Comment:
are you able to do this with CSS transitions and animations instead?

Reply:
Yeah sorta. Not exactly exit effects... I was planning this for a later update though since its complex to make for exiting. In my opinion its best to leave this as is until the next update rather than rushing and pushing it out... right?

Reply:
Screw it, Im just gonna do it now.